sda/cmd/verify/verify.go

```diff
@@ -345,60 +345,56 @@ func main() {
 
 					continue
 				}
-				switch status {
-				case "disabled":
+
+				if status == "disabled" {
 					log.Infof("file with correlation ID: %s is disabled, stopping verification", delivered.CorrelationId)
 					if err := delivered.Ack(false); err != nil {
 						log.Errorf("Failed acking canceled work, reason: (%s)", err.Error())
 					}
 
 					continue
-				case "enabled":
-					fileInfo, err := db.GetFileInfo(message.FileID)
-					if err != nil {
-						log.Errorf("failed to get info for file: %s", message.FileID)
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
+				}
 
-						continue
+				fileInfo, err := db.GetFileInfo(message.FileID)
+				if err != nil {
+					log.Errorf("failed to get info for file: %s", message.FileID)
+					if err := delivered.Nack(false, true); err != nil {
+						log.Errorf("failed to Nack message, reason: (%s)", err.Error())
 					}
 
-					if fileInfo.DecryptedChecksum != "" {
-						log.Debugln("file already verified")
-						if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, conf.Broker.RoutingKey, verifiedMessage); err != nil {
-							log.Errorf("failed to publish message, reason: (%s)", err.Error())
-							if err := delivered.Nack(false, true); err != nil {
-								log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-							}
+					continue
+				}
 
-							continue
-						}
+				switch {
+				case fileInfo.DecryptedChecksum != "" && fileInfo.DecryptedChecksum == fmt.Sprintf("%x", sha256hash.Sum(nil)):
```

Comment:
```suggestion
				case fileInfo.DecryptedChecksum == fmt.Sprintf("%x", sha256hash.Sum(nil)):
```

The second part can handle all three possible combinations alone.